lib/Dialect/LLHD/Simulator/Engine.cpp

```diff
@@ -139,19 +148,24 @@ int Engine::simulate(int n) {
           // Invalidate scheduled wakeup
           state->instances[inst].expectedWakeup = Time();
         }
-        wakeupQueue.insert(inst);
+        wakeupQueue.push_back(inst);
       }
 
       // Dump the updated signal.
-      state->dumpSignal(out, change.first);
+      if (traceMode >= 0)
+        trace.addChange(change.first);
```

Comment:
I like the new tracer API and the fact that it is optional :+1: All the speed! :dash: